includes/class-wc-abstract-google-analytics-js.php

```diff
@@ -107,6 +107,20 @@ public function add_transaction( $order ) {
 	 */
 	abstract protected function add_transaction_enhanced( $order );
 
+	/**
+	 * Get item identifier from product data
+	 *
+	 * @param  WC_Product $product WC_Product Object
+	 * @return stirng
+	 */
+	public static function get_product_identifier( $product ) {
```

Comment:
You're right that `protected` would have worked fine here, however, in the last update it is now being called from [class-wc-google-analytics.php](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/2754e1955f0779f9e090eaa6df1cc4ea3cb8635d/includes/class-wc-google-analytics.php#L492) so it will now need to remain public.

Reply:
In my opinion the `add_to_cart` event code is badly placed in the main class. As far as I can see it's outputting a JS snippet just like the rest of the events. But that's outside the scope of this PR, so fine to leave as is.